Build Teleport Connect for Linux

zmb3 opened the pull request:
In order to do so, we add a new make target:

    make teleterm

This (temporarily) assumes that the gravitational/webapps repo is
cloned at the right version as a sibling to the teleport repo.
(We'll be able to get rid of this when we merge webapps into Teleport)

Additionally, update dronegen to include the name of the calling
function that generated the snippet instead of the line number.
This gets rid of lot's of superfluous diffs in the generated
.drone.yml file.

Co-authored-by: Grzegorz Zdunek <grzegorz.zdunek@goteleport.com>

gzdunek commented on dronegen/tag.go:
```diff
@@ -148,6 +176,12 @@ func tagCopyArtifactCommands(b buildType) []string {
 
+	if b.hasTeleportConnect() {
+		commands = append(commands,
+			`cp /go/src/github.com/gravitational/webapps/packages/teleterm/build/release/teleconnect*.{tar.gz,deb,rpm} /go/artifacts/`,
```
We just changed `teleconnect` to `teleport-connect` in webapps.
```suggestion
			`cp /go/src/github.com/gravitational/webapps/packages/teleterm/build/release/teleport-connect*.{tar.gz,deb,rpm} /go/artifacts/`,

zmb3 replied:
Does this affect the Mac builds too?

gzdunek replied:
No, this change affects only Linux packages.

gzdunek reviewed (commented):
Look fine! 

I tried to run it on Drone but it failed with an error:
```
git checkout $(go run $WORKSPACE_DIR/go/src/github.com/gravitational/teleport/build.assets/tooling/cmd/get-webapps-version/main.go)
/bin/sh: go: not found
```

Also I would wait with merging this PR until https://github.com/gravitational/webapps/pull/1098 is merged.

zmb3 commented:
> Look fine!
> 
> I tried to run it on Drone but it failed with an error:
> 
> ```
> git checkout $(go run $WORKSPACE_DIR/go/src/github.com/gravitational/teleport/build.assets/tooling/cmd/get-webapps-version/main.go)
> /bin/sh: go: not found
> ```

I see. We don't have Go yet because this step is not running inside docker. Hm...

zmb3 committed:
Build Teleport Connect for Linux

In order to do so, we add a new make target:

    make teleterm

This (temporarily) assumes that the gravitational/webapps repo is
cloned at the right version as a sibling to the teleport repo.
(We'll be able to get rid of this when we merge webapps into Teleport)

Additionally, update dronegen to include the name of the calling
function that generated the snippet instead of the line number.
This gets rid of lots of superfluous diffs in the generated
.drone.yml file.

Co-authored-by: Grzegorz Zdunek <grzegorz.zdunek@goteleport.com>

zmb3 committed:
Add make target to enter the teleterm build container

zmb3 committed:
Update Teleport Connect artifact name

zmb3 committed:
Rewrite get-webapps-version in bash

The original thought was that Go would be easier to test and reason
about, but due to the our drone setup works, requiring Go/Docker at
this step of the process introduced extra complexity.

This script is pretty simple and will be removed later this quarter
when we migrate webapps into the Teleport repo.

zmb3 commented:
We're close. The build pipeline is working now, and we get an error on copying artifacts:

```
cp: can't stat '/go/src/github.com/gravitational/webapps/packages/teleterm/build/release/teleport-connect*.{tar.gz,deb,rpm}': No such file or directory
```

gzdunek commented on build.assets/Dockerfile-teleterm:
```diff
@@ -17,9 +17,14 @@ RUN BIN="/usr/local/bin" && \
+     curl -o /tmp/nodejs.tar.xz -L ${NODE_URL} && \
+     tar -xJf /tmp/nodejs.tar.xz -C /usr/local/lib/nodejs-linux --strip-components=1)
+RUN corepack enable yarn
 
```
We need to add `rpm` tools to build `rpm`s:
```
# Install rpmbuild
RUN apt-get update -y && apt-get install -q -y --no-install-recommends rpm
```

gzdunek commented on build.assets/Makefile:
```diff
@@ -227,6 +232,24 @@ buildbox-teleterm: buildbox
+
+CONNECT_VERSION ?= $(VERSION)
+ifeq ($(CONNECT_VERSION),)
+CONNECT_VERSION := $(BUILDBOX_VERSION)-dev
```
I'm not sure about it, for the other artifacts we have a tag name:
<img width="403" alt="image" src="https://user-images.githubusercontent.com/20583051/185949833-8db77c9e-c68d-474b-b694-a6e68222d19c.png">

gzdunek commented on dronegen/tag.go:
```diff
@@ -148,6 +175,12 @@ func tagCopyArtifactCommands(b buildType) []string {
 
+	if b.hasTeleportConnect() {
+		commands = append(commands,
+			`cp /go/src/github.com/gravitational/webapps/packages/teleterm/build/release/teleport-connect*.{tar.gz,deb,rpm} /go/artifacts/`,
```
The wildcard is a problem for `cp`, we can use `find`:
```
`find /go/src/github.com/gravitational/webapps/packages/teleterm/build/release -maxdepth 1 \( -iname "teleport-connect*.tar.gz" -o -iname "teleport-connect*.rpm" -o -iname "teleport-connect*.deb" \) -print -exec cp {} /go/artifacts/ \;`
```

gzdunek reviewed (commented):
As I wrote on Slack the `cp` was also failing, because the build was done for some old version of webapps (so the artifact name was `teleconnect`, not `teleport-connect`).

zmb3 committed:
Address review comments

- make sure rpm tools are installed
- ensure that version is passed correctly
- use find to identify the binaries to copy
